src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt

```diff
@@ -0,0 +1,230 @@
+    private val skinAccentFill: ObjectProperty<Color> = SimpleObjectProperty<Color>(Color.RED)
+    private val skinTrackColor: ObjectProperty<Color> = SimpleObjectProperty<Color>(Color.WHITE)
+    private val barHeight: DoubleProperty = SimpleDoubleProperty(0.0)
+    private val indicatorRadius: DoubleProperty = SimpleDoubleProperty(0.0)
+
+
+    init {
+        //need to define super(statusindicator)? not sure if this is required atm
+        bar = StackPane()
+        bar.styleClass.setAll("bar")
+
+        track = StackPane()
+        track.styleClass.setAll("track")
+
+        children.setAll(track, bar)
+
+        control.widthProperty().onChange { observable -> invalidBar = true }
+        control.heightProperty().onChange { observable -> invalidBar = true }
+        control.primaryFillProperty.onChange { observable -> updateBarFill(control.primaryFill, control.accentFill) }
+        control.accentFillProperty.onChange { observable -> updateBarFill(control.primaryFill, control.accentFill) }
+        control.progressProperty.onChange { progress ->
+            updateStatusIndicator(control.width, control.height)
+        }
+        skinnable.requestLayout()
+
+        localProgressProperty.bind(control.progressProperty)
+        skinPrimaryFill.bind(control.primaryFillProperty)
+        skinAccentFill.bind(control.accentFillProperty)
+        barHeight.bind(control.barHeightProperty)
+        indicatorRadius.bind(control.indicatorRadiusProperty)
+        skinTrackColor.bind(control.trackFillProperty)
+    }
+
+    fun updateBarFill(primaryFill: Color, accentFill: Color) {
+        val stops = listOf(Stop(0.0, primaryFill), Stop(1.0, accentFill))
+        if (bar != null && track != null) {
+            bar.background = Background(
+                BackgroundFill(
+                    LinearGradient(
+                        0.0,
+                        0.0,
+                        0.05,
+                        0.7,
+                        true,
+                        CycleMethod.REPEAT,
+                        stops
+                    ),
+                    CornerRadii(0.0),
+                    Insets(0.0)
+                )
+            )
+            skinnable.requestLayout()
+        }
+    }
+
+    fun updateStatusIndicator(width: Double, height: Double) {
+        val stops = listOf(Stop(0.0, skinPrimaryFill.value), Stop(1.0, skinAccentFill.value))
+
+        if (bar != null && track != null) {
+            children.remove(bar)
+            children.remove(track)
+        }
+
+        bar = StackPane()
+        track = StackPane()
+        if (localProgressProperty.value <= 1.0000001) {
+            barWidth =
+                ((localProgressProperty.value * width) - snappedLeftInset() - snappedRightInset()).toInt().toDouble() * 2.0 * Math.min(
+                    1.0,
+                    Math.max(0.0, 1.0)
+                ) / 2.0f
```

Comment:
`2.0 * Math.min(1.0, Math.max(0.0, 1.0)) / 2.0f`
Can be simplified.

Reply:
Line 224 below too.

Reply:
Math.max(0.0, 1.0) = 1.0
Math.min(1.0, 1.0) = 1.0
2.0 * 1.0 = 2.0
2.0/2.0 = 1.0

What was the point of this at all? There's no variables here?

Reply:
It was something that I pulled from the javafx progressbar for calculation the barWidth. I didn't really give it to much attention after I put it in there